scanpy/tests/test_clustering.py

```diff
@@ -186,6 +189,22 @@ def test_louvain_basic(adata_neighbors):
     sc.tl.louvain(adata_neighbors, flavor="igraph")
 
 
+@needs.louvain
+@pytest.mark.parametrize("random_state", [10, 999])
+@pytest.mark.parametrize("resolution", [0.9, 1.1])
+def test_louvain_custom_key(adata_neighbors, resolution, random_state):
+    sc.tl.louvain(
+        adata_neighbors,
+        key_added="louvain_custom",
+        random_state=random_state,
+        resolution=resolution,
+    )
+    assert (
+        adata_neighbors.uns["louvain_custom"]["params"]["random_state"] == random_state
+    )
+    assert adata_neighbors.uns["louvain_custom"]["params"]["resolution"] == resolution
```

Comment:
I'm don't completely understand the details of how codecov works, but I think it fails to pick up this test. Apologies if I'm doing something wrong!

Reply:
No worries, codecov can be flaky.  Can we paramterize this test to do both `leiden` and `louvain` and then also run it twice with two or three (including default) different keys so as to check that we're not doing overriding unintentionally?

Reply:
Good idea. I'll add that.